STYLEGUIDE.md

```diff
@@ -0,0 +1,367 @@
+# Design Systems Style Guide
+
+## Table of Contents
+
+- [To Contribute](#To-Contribute)
+- [Typing](#Typing)
+- [Functions](#Functions)
+- [Variable naming](#Variable-naming)
+- [Variable placement](#Variable-placement)
+- [Whitespace and line breaks](#Whitespace-and-line-breaks)
+- [JSX/React](#JSX-React)
+- [CSS-in-JS](#CSS-in-JS)
+- [File Structure](https://github.com/mongodb/leafygreen-ui/blob/main/stories/Folder-Structure.stories.mdx)
+- [References](#References)
+
+## <a id="To-Contribute">To Contribute</a>
+
+To propose updates to this style guide, please make the relevant changes in a branch and submit a PR. We review these PRs synchronously in our weekly Team Code Review and collectively gavel on new standards there.
+
+### Format Contributions
+
+### Prefer [Rule]
+
+#### Why
+
+_Description_
+
+#### Prefer
+
+_Example_
+
+#### Avoid
+
+_Example_
+
+---
+
+## <a id="Typing">Typing</a>
+
+### Prefer constants and types over enums.
+
+#### Why
+
+Ensures that TypeScript doesn't prevent people from using the values directly rather than consuming the enum.
+
+#### Prefer
+
+```typescript
+export const someConstant = {
+  optionOne = 'option 1',
+  optionTwo = 'option 2',
+} as const;
+```
+
+#### Avoid
+
+```typescript
+enum someConstant = {
+ optionOne: 'option 1',
+ optionTwo: 'option 2',
+}
+```
+
+---
+
+### When creating TS entities, place them at the top of the file, above JS
```

Comment:
Why?

Reply:
Helps keeps files organized, and if we want to know where to look for things this will be helpful